design/ceph/mon-storage-class.md

```diff
@@ -0,0 +1,136 @@
+# Mon Multiple Storage Class Support
+
+## Summary
+
+It is generally desirable in production to have multiple zones availability. A
+whole storage class could go unavailable then and enough mons could stay up to
+keep the cluster running.
+In such a case the distribution of mons can be made more homogeneous for each
+zone, by this we can leverage the spread of mons for providing high
+availability. This could be a useful scenario for a datacenter with own storage
+class in contrast to ones provided by cloud providers.
+
+Rook will enable the mon deployment when requested by the admin through the
+usual Rook CRDs.
+
+## Architecture
+
+To support this scenario, we can do something similar to the mon deployment for
+[Stretch clusters](https://github.com/ceph/ceph/blob/master/doc/rados/operations/stretch-mode.rst).
+The operator will have zonal awareness, associating mon deployments based on
+zones configuration provided.
+
+The type of failure domain used for this scenario is commonly "zone", but can
+be set to a different failure domain.
+
+### Failure Domain Configuration
+
+The topology of the K8s cluster is to be determined by the admin, outside the scope of Rook.
+Rook will simply detect the topology labels that have been added to the nodes.
+
+If the desired failure domain is a "zone", the `topology.kubernetes.io/zone` label should
+be added to the nodes. Any of the [topology labels](https://rook.io/docs/rook/latest/ceph-cluster-crd.html#osd-topology)
+supported by OSDs can be used.
+
+For example:
+
+```yaml
+topology.kubernetes.io/zone=a
+topology.kubernetes.io/zone=b
+topology.kubernetes.io/zone=c
+```
+
+### Rook Cluster Design
+
+The core changes to rook are to associate the mons with the required zones.
+
+- Each zone will have at least one mon assigned using labels for nodeAffinity.
+
+The new configuration will support the `zones` configuration where the
+failure domains available must be listed and the arbiter zone is identified
+optionally for mon configuration.
+
+```yaml
+  mon:
+    count: 3
+    allowMultiplePerNode: false
```

Comment:
This should always be `false` for this use case.  The code should handle this somewhere, IMO.

Reply:
I believe we will find allowing multiple per node is independent of implementation of this feature